### ISSUE TITLE
The UIAHandler.utils.getDeepestLastChildUIAElementInWalker function don't work as expected

### ISSUE THREAD

**Issue by Emil-18** [nvaccess/nvda#15101]
### Steps to reproduce:
* Open for example the windows 10 calculator, and set the navigator object to the Number pad grouping
* open the python console with CTRL+Z and call the function with nav.UIAElement and UIAHandler.handler.baseTreeWalker. 
### Actual behavior:

The UIAElement that it returns is the last child of the nav.UIAElement, instead of the deepest last child. This happens because the function returns in the while loop instead of outside of it
### Expected behavior:

It should return the deepest last child like it says it does
### NVDA logs, crash dumps and other attachments:
N/A
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
alpha-28584,5c4e6e1f (2023.2.0.28584)
#### Windows version:
N/A
#### Name and version of other software in use when reproducing the issue:
N/A
#### Does the issue still occur after restarting your computer?
N/A
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
N/A

**Comment by seanbudd** [comment]
Is there a known bug related to this?

**Comment by Emil-18** [comment]
None that I know of. I found the bug because I was intending to use the function in an add-on I am working on

**Comment by seanbudd** [comment]
Ah - I believe the behaviour of this function is intentional, based on how it is used.
Perhaps it should be named and documented better, and it probably should be marked as private with a preceding underscore.

**Comment by Emil-18** [comment]
I don't think it is intentional, because as it is now, it does the exact same thing as walker.GetLastChildElement. Plus, if it were working as expected, why would it enter a while loop instead of just returning the last child element?

**Comment by seanbudd** [comment]
Are you certain these functions are the same?
Without a production bug with the code - we can assume that is is working as expected.
The behaviour of the function is just different to what you assume it to be - likely a naming or documentation problem.
It would be risky to change the behaviour of the function to match the naming unless it solves an actual problem with NVDA.

**Comment by Emil-18** [comment]
I checked by making a list containing all current UIAElements on the system using UIAElement.FindAll. I then did the following
for i in List:
	e1 = walker.GetLastChildElement(i)
	e2 = UIAHandler.utils.getDeepestLastChildUIAElementInWalker(i, walker)
	if e1 and e2 and not UIAHandler.handler.clientObject.CompareElements(e1, e2):
		print('e')
I didn't get any output, so to me at least, they seam identical

**Comment by seanbudd** [comment]
I think we would need fairly thorough unit tests with different inputs to confirm the functions are the same

**Comment by seanbudd** [comment]
Regardless - if the functions are the same, it is still not a good idea to change the behaviour of a working piece of code.

**Comment by Emil-18** [comment]
I see. The issue is very easy to work around, so it isn't actually a big problem at all. I just think that the function should do what it says it does
